discovery_engine_core/bindings/src/tracing.rs

```diff
@@ -14,37 +14,105 @@
 
 //! Setup tracing on different platform.
 
-use std::sync::Once;
+use std::{env::VarError, error::Error, fs::OpenOptions, io, path::Path, sync::Once};
 
-use tracing_subscriber::{filter::LevelFilter, util::SubscriberInitExt};
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
 
 static INIT_TRACING: Once = Once::new();
 
-pub(crate) fn init_tracing() {
+pub(crate) fn init_tracing(log_file: Option<&Path>) {
     INIT_TRACING.call_once(|| {
-        init_tracing_once();
+        init_tracing_once(log_file);
+        init_panic_logging();
     });
 }
 
-fn init_tracing_once() {
-    let subscriber = tracing_subscriber::fmt()
-        .with_max_level(LevelFilter::INFO)
-        .finish();
+fn init_tracing_once(log_file: Option<&Path>) {
+    let mut delayed_errors = Vec::new();
 
+    let stdout_log = tracing_subscriber::fmt::layer();
+
+    #[allow(clippy::needless_late_init)]
+    let android_logging;
```

Comment:
🔧 the `clippy::needless_late_init` can be avoided by moving the binding inside the macro. (i can't mark it as a suggestion because of the multiline diff)
```rust
    cfg_if::cfg_if! {
        if #[cfg(target_os = "android")] {
            let android_logging = tracing_android::layer("xayn_discovery_engine").ok();
        } else {
            // workaround to not have to specify a alternative type per hand
            let android_logging = false.then(tracing_subscriber::fmt::layer);
        }
    };
```